CoJezdzi/AppFramework/_Redux/_Middleware/AppMiddleware.swift

```diff
@@ -1,5 +1,66 @@
-
 import ReSwift
+import APIKit
+import RxSwift
+import RxCocoa
+
+struct FetchAction: Action {
+    let request: UMWarsawTransportApi.GetRealtimeDataRequest
+    let resultFactory: ([WarsawVehicleDto]) -> Action
+}
+
+struct AppMiddleware {
+    
+    static let Api: Middleware<AppState> = { dispatch, getState in
+        return { next in
+            return { action in
+                guard action is FetchVehiclesPosytionsAction, let state = getState() else {
+                    next(action)
+                    return
+                }
+                
+                fetchActions(for: state.settingsState)
+                    .compactMap { action in
+                        let request = action.request
+                        networkSession.rx.response(request).compactMap { result in
+                            action.resultFactory(result)
+                        }
+                    }
+                    .observeOn(MainScheduler.instance)
+                    .subscribe(onNext: { action in
+                        dispatch(action)
+                    })
+            }
+        }
+    }
+    
+    fileprivate static var networkSession: Session {
+        return Session.shared
```

Comment:
All this `shared` stuff should be moved to the Environment structure :) It's easer to mock it in tests and in playgrounds. That said could you also provide a mock implementation (it could return empty result) in `TestsHelpers->_Mocks` :)